rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/InlineItemModificationCreator.java

```diff
@@ -0,0 +1,171 @@
+
+        nodeSourceWindow.addCredentialsPickerIcon(pluginField,
+                                                  formItemsReplacingNonTextualFormItems,
+                                                  chooseCredentialsFormItem);
+
+        return formItemsReplacingNonTextualFormItems;
+    }
+
+    private FormItem createUploadItemDisabled(PluginDescriptor plugin, PluginDescriptor.Field pluginField) {
+
+        FormItem chooseCredentialsFormItem;
+        chooseCredentialsFormItem = new UploadItem(plugin.getPluginName() + pluginField.getName(), "");
+        chooseCredentialsFormItem.disable();
+        return chooseCredentialsFormItem;
+    }
+
+    private TextAreaItem createTextItemPrefilledWithFileContent(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+
+        TextAreaItem previousValueItem = new TextAreaItem(plugin.getPluginName() + pluginField.getName() +
+                                                          EDIT_FORM_ITEM_SUFFIX, "");
+
+        previousValueItem.setDefaultValue(pluginField.getValue());
+
+        return previousValueItem;
+    }
+
+    private RadioGroupItem createRadioItemToModifyPluginField(PluginDescriptor plugin,
+            PluginDescriptor.Field pluginField) {
+
+        String formItemPrefixName = plugin.getPluginName() + pluginField.getName();
+
+        LinkedHashMap<String, String> radioOptions = new LinkedHashMap<>();
+        radioOptions.put(EDIT_RADIO_OPTION_NAME, "In-line editing");
+        radioOptions.put(UPLOAD_RADIO_OPTION_NAME, "Upload new file");
+
+        RadioGroupItem editOrUploadFormItem = new RadioGroupItem(formItemPrefixName + EDIT_OR_UPLOAD_FORM_ITEM_SUFFIX,
+                                                                 pluginField.getName());
+        editOrUploadFormItem.setValueMap(radioOptions);
+        editOrUploadFormItem.setVertical(false);
+        editOrUploadFormItem.setDefaultValue(EDIT_RADIO_OPTION_NAME);
+
+        editOrUploadFormItem.addChangedHandler(changedEvent -> {
+
+            String radioValue = changedEvent.getValue().toString();
+            List<FormItem> formItemsForPlugin = this.nodeSourceWindow.getFormItemsOfPlugin(plugin.getPluginName());
+
+            if (radioValue.equals(EDIT_RADIO_OPTION_NAME)) {
+                enableEditInLine(formItemPrefixName, formItemsForPlugin);
+            } else if (radioValue.equals(UPLOAD_RADIO_OPTION_NAME)) {
+                enableUploadNewFile(formItemPrefixName, formItemsForPlugin);
+            }
+        });
+
+        return editOrUploadFormItem;
+    }
+
+    private void enableUploadNewFile(String pluginFieldName, List<FormItem> formItemsForPlugin) {
+
+        formItemsForPlugin.stream()
+                          .filter(formItem -> formItem.getName().startsWith(pluginFieldName))
+                          .forEach(formItem -> enableUploadNewFile(pluginFieldName, formItem));
+    }
+
+    private void enableUploadNewFile(String pluginFieldName, FormItem formItem) {
```

Comment:
can you rename this method to enableOrDisableUploadNewFile for clarity?

Reply:
Done.